src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -564,7 +564,8 @@ event::ptr primitive_inst::realloc_if_needed() {
 
 
     // If we allocated too large memory, reclaim the memory.
-    if (updated_layout.get_buffer_size().count() * 10 < _max_output_layout_count) {
+    if (get_network().get_config().get_property(ov::intel_gpu::hint::aggressive_memory_opt)
```

Comment:
Please save this flag somewhere. Query of the property on each iteration may be expensive.

Reply:
Applied